src/build.ts

```diff
@@ -252,15 +254,28 @@ export async function build(
   }
 
   // Render pages!
+  const buildManifest: BuildManifest = {pages: []};
   for (const [sourceFile, {page, resolvers}] of pages) {
     const sourcePath = join(root, sourceFile);
     const outputPath = join(dirname(sourceFile), basename(sourceFile, ".md") + ".html");
     const path = join("/", dirname(sourceFile), basename(sourceFile, ".md"));
     effects.output.write(`${faint("render")} ${sourcePath} ${faint("→")} `);
     const html = await renderPage(page, {...config, path, resolvers});
     await effects.writeFile(outputPath, html);
+    const url =
+      outputPath === "index.html"
+        ? "/"
+        : basename(outputPath) === "index.html"
+        ? join("/", dirname(outputPath), "/")
+        : extname(outputPath) === ".html"
+        ? join("/", dirname(outputPath), basename(outputPath, ".html"))
+        : outputPath;
```

Comment:
Is there code already in Framework that does this? It looks a lot like `getPathNormalizer` in `config.ts` (though not exactly). Maybe it’s fine to duplicate, just wondering if there’s an opportunity to consolidate.

Reply:
`getPathNormalizer` was very nearly what I wanted, including the clean url handling which I didn't have in this version. The only difference was that it produced a path of `"."` for a root index.html. Prefixing `outputPath` with `"/"` fixed that though.